scrapy_poet/page_input_providers.py

```diff
@@ -179,10 +179,10 @@ def __call__(self, to_provide: Set[Callable], response: Response):
             )
         ]
 
-    def fingerprint(self, to_provide: Set[Callable], request: Request) -> str:
+    def fingerprint(self, to_provide: Set[Callable], request: Request, spider: Spider) -> str:
         request_keys = {"url", "method", "body"}
         request_data = {
-            k: str(v) for k, v in request.to_dict().items() if k in request_keys
+            k: str(v) for k, v in request.to_dict(spider=spider).items() if k in request_keys
```

Comment:
Given the callback is not actually meant to be stored by any of these methods, I am thinking creating a copy of the original request without callback or errback, and using that for serialization purposed, may be best here.

e.g. (untested)

```
_request = request.replace(callback=None, errback=None)
```

Then you use `_request` in the rest of the method instead of `request`.

Reply:
OK. Makes sense I will make that change. @Gallaecio However, I will still have to pass ```spider``` to the ```to_dict``` method, am I right? In that case do you recommend the argument style used here, or should I use the crawler method passed to the parent object?

Reply:
> I will still have to pass spider to the to_dict method, am I right?

[The spider argument is only needed for requests with a callback or errback](https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.to_dict), so the idea is not to need to pass it at all after you set those to `None`. I am hoping there will be no exception provided those methods are set to `None`.

Reply:
Yes, thank you for the suggestion. I have made this change now.